src/backend/joanie/payment/models.py

```diff
@@ -344,12 +344,31 @@ def save(self, *args, **kwargs):
         super().save(*args, **kwargs)
 
 
+class CreditCardManager(models.Manager):
+    """Custom manager for `CreditCard` model"""
+
+    def get_card_for_owner(self, token, user_id):
+        """
+        Retrieve a main credit card for a given owner.
+        This method attempts to retrieve a credit card that matches the given `token` and
+        `user_id`, and is marked as the main card. If no such card exists, it returns None.
+        """
+        try:
+            return self.get(token=token, owner_id=user_id, is_main=True)
```

Comment:
Should we force the card to be the main one ?

Reply:
I think we might need the thought of @jbpenrath. I've always thought we would use the main card.

Reply:
Nope a user can have several credit cards. It should be able choose to use another one than the main one.

Reply:
I will leave this review open until our friend gives his thought :wink:

Reply:
Okay, I will take away `is_main` from the get queryset